ui/src/react-admin/modules/content-page/components/blocks/BlockContentEnclose/hooks/useGetContentBlockEncloseContent.ts

```diff
@@ -55,31 +61,53 @@ export const useGetContentBlockEncloseContent = (ids: MappedElement[]): Enclosed
 		queries: [...(ieObjectIds.length > 0 ? [ieObjectQuery] : []), ...contentPageQueries],
 	});
 
-	return compact(results).flatMap((result: any) => {
-		if (!result.data) {
-			return null;
-		}
-		if (Array.isArray(result.data)) {
-			return result.data.map((item: any) => {
-				return {
-					id: item.maintainerId,
-					name: item.name,
-					description: item.description,
-					thumbnail: item.thumbnailUrl,
-					dateCreated: item.dateCreatedLowerBound,
-					maintainerName: item.maintainerName,
-					icon: item.ebucoreObjectType,
-					type: 'IE_OBJECT',
-				};
-			});
+	const mappedResults = compact(results).flatMap((result) => {
+		if (result.status === 'success') {
+			if (!result.data) {
+				return null;
+			}
+			if (Array.isArray(result.data)) {
+				const ieObjects = result.data as IeObject[];
+				return ieObjects.map((item: IeObject) => {
+					return {
+						id: item.maintainerId,
+						name: item.name || item.title,
+						description: item.description,
+						thumbnail:
+							item.thumbnailUrl ||
+							'https://fastly.picsum.photos/id/716/600/600.jpg?hmac=rfg-0QfaSm9tSWioKwljrGVZIXvIy_KZuuFZO0H7bAQ',
+						dateCreated: item.dateCreatedLowerBound,
+						maintainerName: item.maintainerName,
+						objectType: item.dctermsFormat,
+						identifier: item.schemaIdentifier,
+						pid: item.meemooIdentifier,
+						link: `/zoeken/maintainer-slug/${item.schemaIdentifier}/${kebabCase(
+							item.name
+						)}`,
+						type: 'IE_OBJECT',
+					};
+				});
+			}
+
+			const contentPage = result.data as ContentPage;
+			return {
+				id: contentPage.id,
+				name: contentPage.title,
+				description: contentPage.description,
+				thumbnail: contentPage.thumbnailPath,
+				identifier: contentPage.path,
+				link: contentPage.path,
+				type: 'CONTENT_PAGE',
+			};
 		}
+	});
 
-		return {
-			id: result.data.id,
-			name: result.data.title,
-			description: result.data.description,
-			thumbnail: result.data.thumbnailPath,
-			type: 'CONTENT_PAGE',
-		};
-	}) as any;
+	return compact(
+		originalElements.map((element) => {
+			const found = mappedResults.find(
+				(item) => (item && item.identifier === element?.mediaItem?.value) || null
```

Comment:
In a find function you need to return a truthy value
no no need to default to null, since the first part will already be falsy
```suggestion
				(item) => item && item.identifier === element?.mediaItem?.value
```